lume_model/models/torch_model.py

```diff
@@ -127,6 +123,19 @@ def evaluate(
         output_dict = self._prepare_outputs(parsed_outputs)
         return output_dict
 
+    def input_validation(self, input_dict: dict[str, Union[float, torch.Tensor]]):
+        """Itemizes tensors before performing input validation."""
+        formatted_inputs = self._format_inputs(input_dict)
+        itemized_inputs = self._itemize_dict(formatted_inputs)
```

Comment:
We are calling these methods expecting to be passing a dict with floats or torch tensors as the values before having validated them. In cases where we have the wrong type, say a bool, string, or `None`, it won't behave as expected.